pkg/controller.v2/controller_status.go

```diff
@@ -132,11 +260,14 @@ func newCondition(conditionType tfv1alpha2.TFJobConditionType, reason, message s
 
 // getCondition returns the condition with the provided type.
 func getCondition(status tfv1alpha2.TFJobStatus, condType tfv1alpha2.TFJobConditionType) *tfv1alpha2.TFJobCondition {
-	for i := range status.Conditions {
+	/*for i := range status.Conditions {
 		c := status.Conditions[i]
 		if c.Type == condType {
 			return &c
 		}
+	}*/
+	if len(status.Conditions) > 0 {
+		return &status.Conditions[len(status.Conditions)-1]
```

Comment:
Why do we return the last condition instead of return the condition with the provided type? We are using the function in setCondition and the change will break it, IMO.

Reply:
I think that the state should be created-> running-> failed. The final state is determined in the order of the array, but the previous method is, for example: the current state is running, but the array form is created->running->failed. Personally think this is not very good